[REEF-182] Clean up Link and LinkListener for better message and error handling

gwsshs22 opened the pull request:
This addressed issue by

    * Removing throw clause of write method in Link interface
    * Modifying LinkListener interface and implemented classes to work properly in asynchronous communication
    * Removing redundant try-catches where Link instances were used before

JIRA : https://issues.apache.org/jira/browse/REEF-182

Author : Geon Woo Kim <gwsshs22@gmail.com>

bgchun commented on lang/java/reef-io/src/main/java/org/apache/reef/io/network/impl/NSConnection.java:
```diff
@@ -129,6 +125,10 @@ public void close() throws NetworkException {
 
   @Override
-  public void messageReceived(final NSMessage<T> message) {
+  public void onSuccess(NSMessage<T> message) {
```
final

bgchun commented on lang/java/reef-io/src/main/java/org/apache/reef/io/network/impl/NSConnection.java:
```diff
@@ -129,6 +125,10 @@ public void close() throws NetworkException {
+  }
+
+  @Override
+  public void onException(Throwable cause, SocketAddress remoteAddress, NSMessage<T> message) {
```
final

bgchun commented on lang/java/reef-io/src/main/java/org/apache/reef/io/network/impl/NetworkService.java:
```diff
@@ -206,7 +207,13 @@ public void close() throws Exception {
         this.myId, destId, new LinkListener<T>() {
       @Override
-      public void messageReceived(final Object message) {
+      public void onSuccess(T message) {
```
final

bgchun commented on lang/java/reef-io/src/main/java/org/apache/reef/io/network/impl/NetworkService.java:
```diff
@@ -206,7 +207,13 @@ public void close() throws Exception {
       @Override
-      public void messageReceived(final Object message) {
+      public void onSuccess(T message) {
+
```
remove the line

bgchun commented on lang/java/reef-io/src/main/java/org/apache/reef/io/network/impl/NetworkService.java:
```diff
@@ -206,7 +207,13 @@ public void close() throws Exception {
+      }
+
+      @Override
+      public void onException(Throwable cause, SocketAddress remoteAddress, T message) {
```
final

bgchun replied:
Need at least logging

bgchun commented on lang/java/reef-io/src/main/java/org/apache/reef/io/network/impl/NetworkService.java:
```diff
@@ -206,7 +207,13 @@ public void close() throws Exception {
+
+      @Override
+      public void onException(Throwable cause, SocketAddress remoteAddress, T message) {
+
```
remove this line

bgchun commented on lang/java/reef-io/src/main/java/org/apache/reef/io/network/naming/NameRegistryClient.java:
```diff
@@ -145,7 +145,11 @@ public void unregister(Identifier id) throws IOException {
         new LinkListener<NamingMessage>() {
           @Override
-          public void messageReceived(NamingMessage message) {
+          public void onSuccess(NamingMessage message) {
```
final

bgchun commented on lang/java/reef-io/src/main/java/org/apache/reef/io/network/naming/NameRegistryClient.java:
```diff
@@ -145,7 +145,11 @@ public void unregister(Identifier id) throws IOException {
+          }
+
+          @Override
+          public void onException(Throwable cause, SocketAddress remoteAddress, NamingMessage message) {
```
final

bgchun replied:
Need at least logging.

bgchun commented on lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/remote/transport/netty/LoggingLinkListener.java:
```diff
@@ -33,14 +34,22 @@
-      LOG.log(Level.FINEST, "The linklistener " + this.getClass().toString() + "has received " + message);
+  public void onSuccess(T message) {
+    if (LOG.isLoggable(Level.FINEST)) {
+      LOG.log(Level.FINEST, "The message is successfully sent : " + message);
```
Use the MessageFormat style

bgchun commented on lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/remote/transport/netty/LoggingLinkListener.java:
```diff
@@ -33,14 +34,22 @@
+  @Override
+  public void onException(Throwable cause, SocketAddress remoteAddress, T message) {
+    if (LOG.isLoggable(Level.FINEST)) {
+      LOG.log(Level.FINEST, "The message to " + remoteAddress + " is failed to be sent : " + message + ", cause : " + cause);
```
Use the MessageFormat style

bgchun commented:
@gwsshs22 I left some comments.

gwsshs22 commented:
Hi,

Could you  teach me how can I address your feedback?
This is first time to me contributing open source project, so I has
difficulty on addressing feedback on submitted pull request.

Should I make another commit and new pull request or push more commit this
pull request?

Thank you,
Geon Woo Kim.

2015년 3월 1일 일요일, Byung-Gon Chun<notifications@github.com>님이 작성한 메시지:

> @gwsshs22 <https://github.com/gwsshs22> I left some comments.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-reef/pull/96#issuecomment-76593398>.
>

dafrista commented:
Please see https://cwiki.apache.org/confluence/display/REEF/Contributing




In particular, you seem to be on step 7. TLDR: push new commits onto your github branch.


7. The code review
REEF follows a review-then-commit (RTC) model. We perform all our code reviews on GitHub: Community members will leave comments on your pull request and suggest changes. You can have a look at prior pull requests to get an idea of what to expect. During this process, you may want to change your code. You can do that by pushing additional commits to your branch. Don't worry about squashing the commits into one at this stage. This will be done by the committer once your code is merged into REEF.

When the code review concludes, one of Committers will merge your work into the REEF codebase. Good job!

On Sun, Mar 1, 2015 at 9:41 PM, gwsshs22 <notifications@github.com> wrote:

> Hi,
> Could you  teach me how can I address your feedback?
> This is first time to me contributing open source project, so I has
> difficulty on addressing feedback on submitted pull request.
> Should I make another commit and new pull request or push more commit this
> pull request?
> Thank you,
> Geon Woo Kim.
> 2015년 3월 1일 일요일, Byung-Gon Chun<notifications@github.com>님이 작성한 메시지:
>> @gwsshs22 <https://github.com/gwsshs22> I left some comments.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/apache/incubator-reef/pull/96#issuecomment-76593398>.
>>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/apache/incubator-reef/pull/96#issuecomment-76595090

gwsshs22 commented:
I pushed new a commit. Thank you.

bgchun commented:
@gwsshs22 Any chance that NetworkService's need different LinkListener's?

gwsshs22 commented:
I have planned to make user register NetworkExceptionHandlers for each event types. Network service should set one specific LinkListener per link and it will dispatch error callback to users ExceptionHandler.
